pkg/registry/event/rest.go

```diff
@@ -45,7 +46,8 @@ func NewStorage(registry generic.Registry) *REST {
 func (rs *REST) Create(ctx api.Context, obj runtime.Object) (runtime.Object, error) {
 	event, ok := obj.(*api.Event)
 	if !ok {
-		return nil, fmt.Errorf("invalid object type")
+		glog.V(2).Infof("Invalid object type. Was expecting an object of type api.Event: %v", obj)
```

Comment:
For programmer errors where you lose detail, you can also do util.HandleError which glogs by default (but we have extensions to report to services like Sentry or Airbrake).  I need to move those upstream post 1.0.

HandleError is the right target for all programmer errors

> On Jun 24, 2015, at 5:04 PM, Nikhil Jindal <notifications@github.com> wrote:
> 
> In pkg/registry/event/rest.go:
> 
> > @@ -45,7 +46,8 @@ func NewStorage(registry generic.Registry) *REST {
> >  func (rs *REST) Create(ctx api.Context, obj runtime.Object) (runtime.Object, error) {
> >  	event, ok := obj.(*api.Event)
> >  	if !ok {
> > -		return nil, fmt.Errorf("invalid object type")
> > +		glog.V(2).Infof("Invalid object type. Was expecting an object of type api.Event: %v", obj)
> Surfacing internal details in user errors to enable debugging doesnt seem right to me.
> We should return a simple error that indicates something is wrong. We should then be able to figure out the problem, with repro steps, by looking at the logs.
> 
> This is a programmer error. User errors are caught here: https://github.com/GoogleCloudPlatform/kubernetes/blob/141eba39e93ef3f8fe53ce09405a5b83db9737a4/pkg/apiserver/resthandler.go#L305
> 
> —
> Reply to this email directly or view it on GitHub.
>